routers/api/v1/repo/commits.go

```diff
@@ -120,3 +122,204 @@ func GetSingleCommit(ctx *context.APIContext) {
+		baseCommit, err = gitRepo.GetCommit(sha)
+		if err != nil {
+			ctx.ServerError("GetCommit", err)
+			return
+		}
+	}
+
+	// Total commit count
+	commitsCountTotal, err := baseCommit.CommitsCount()
+	if err != nil {
+		ctx.ServerError("GetCommitsCount", err)
+		return
+	}
+
+	pageCount := int(math.Ceil(float64(commitsCountTotal) / float64(git.CommitsRangeSize)))
+
+	// Query commits
+	commits, err := baseCommit.CommitsByRange(page)
+	if err != nil {
+		ctx.ServerError("CommitsByRange", err)
+		return
+	}
+
+	userCache := make(map[string]*models.User)
+
+	apiCommits := make([]*api.Commit, commits.Len())
+
+	i := 0
+	for commitPointer := commits.Front(); commitPointer != nil; commitPointer = commitPointer.Next() {
+		commit := commitPointer.Value.(*git.Commit)
+
+		var apiAuthor, apiCommitter *api.User
+
+		// Retrieve author and committer information
+		cacheAuthor, ok := userCache[commit.Author.Email]
+		if ok {
+			apiAuthor = cacheAuthor.APIFormat()
+		} else {
+			author, err := models.GetUserByEmail(commit.Author.Email)
+			if err != nil && !models.IsErrUserNotExist(err) {
+				ctx.ServerError("Get user by author email", err)
+				return
+			} else if err == nil {
+				apiAuthor = author.APIFormat()
+				userCache[commit.Author.Email] = author
+			}
+		}
+		cacheCommitter, ok := userCache[commit.Committer.Email]
+		if ok {
+			apiCommitter = cacheCommitter.APIFormat()
+		} else {
+			committer, err := models.GetUserByEmail(commit.Committer.Email)
+			if err != nil && !models.IsErrUserNotExist(err) {
+				ctx.ServerError("Get user by committer email", err)
+				return
+			} else if err == nil {
+				apiCommitter = committer.APIFormat()
+				userCache[commit.Committer.Email] = committer
+			}
+		}
+
+		// Retrieve parent(s) of the commit
+		apiParents := make([]*api.CommitMeta, commit.ParentCount())
+		for i := 0; i < commit.ParentCount(); i++ {
+			sha, _ := commit.ParentID(i)
+			apiParents[i] = &api.CommitMeta{
+				URL: ctx.Repo.Repository.APIURL() + "/git/commits/" + sha.String(),
+				SHA: sha.String(),
+			}
+		}
+
+		// Create json struct
+		apiCommits[i] = &api.Commit{
+			CommitMeta: &api.CommitMeta{
+				URL: ctx.Repo.Repository.APIURL() + "/git/commits/" + commit.ID.String(),
+				SHA: commit.ID.String(),
+			},
+			HTMLURL: ctx.Repo.Repository.HTMLURL() + "/commits/" + commit.ID.String(),
+			RepoCommit: &api.RepoCommit{
+				URL: setting.AppURL + ctx.Link[1:],
+				Author: &api.CommitUser{
+					Identity: api.Identity{
+						Name:  commit.Committer.Name,
+						Email: commit.Committer.Email,
+					},
+					Date: commit.Author.When.Format(time.RFC3339),
+				},
+				Committer: &api.CommitUser{
+					Identity: api.Identity{
+						Name:  commit.Committer.Name,
+						Email: commit.Committer.Email,
+					},
+					Date: commit.Committer.When.Format(time.RFC3339),
+				},
+				Message: commit.Summary(),
+				Tree: &api.CommitMeta{
+					URL: ctx.Repo.Repository.APIURL() + "/trees/" + commit.ID.String(),
```

Comment:
This should be `"/git/trees/"` (otherwise is broken link). I know you took it from the getSingleCommit function probably, and that was wrong. Was just fixed in a PR to both develop and release/v1.9

Reply:
fixed